src/TextField/TextFieldInput.js

```diff
@@ -121,7 +121,12 @@ export default class TextFieldInput extends Component {
   }
 
   checkDirty(obj) {
-    if (this.props.onDirty && isDirty(obj)) {
+    const dirty = isDirty(obj);
+    if (dirty === undefined) {
```

Comment:
It seems that controlled vs uncontrolled seems to cause this, this was the visual bug you noted and is present on `next`.   Debugging in-browser, keydown -> `handleChange` -> `checkDirty(this.input)`(good) -> `props.onDirty()` ->`componentWillUpdate` -> `checkDirty(this.props)`(undefined).  I'm unclear on the semantics here of the ref vs componentWillUpdate, so whomever wrote this may want to clear this up.  I'm unsure of the fix here.